mesop/components/table/table.py

```diff
@@ -27,13 +65,19 @@ def _to_data_source(data_frame) -> list[table_pb.TableRow]:
   """Convert Pandas data frame for display as a table.
 
   All values will be converted to strings for display purposes on the frontend.
+
+  The special Pandas `Index` column is included automatically, so we can refer back to
+  same cell on the server to handle user events.
   """
   columns = list(data_frame.columns)
   data = []
   for df_row in data_frame.itertuples():
     data.append(
       table_pb.TableRow(
-        row={col_name: str(getattr(df_row, col_name)) for col_name in columns}
+        row={
+          **{col_name: str(getattr(df_row, col_name)) for col_name in columns},
```

Comment:
There's an edge case here if the one of the columns is called "Index". Pandas seems to rename the column into something else (i.e. _1) to avoid a name collision with the special Index column.

Repro:

https://colab.research.google.com/drive/1TDDNOSFm2jnSEuLwTTqBgPexKUJgr10n#scrollTo=Mizd7VJuzgto

I would add this to table_app.py just to make sure this case is handled correctly.

Reply:
Good catch.

I ended up storing the columns as a repeated list. That should also fix a bug where column names with spaces wouldn't work such as "Date Times".